sys/riotboot/flashwrite.c

```diff
@@ -150,6 +151,34 @@ int riotboot_flashwrite_putbytes(riotboot_flashwrite_t *state,
     return 0;
 }
 
+int riotboot_flashwrite_invalidate(int slot)
+{
+    extern const unsigned riotboot_slot_numof;
+
+    if (riotboot_slot_numof == 1){
+        LOG_WARNING(LOG_PREFIX "Only one slot configured\n");
+        return -1;
+    }
+    if ((riotboot_slot_validate(riotboot_slot_other()) != 0) || (riotboot_slot_validate(riotboot_slot_current()) != 0)){
+        LOG_INFO(LOG_PREFIX "There will be no valid images to run after reboot\n");
+        return -2;
+    }
```

Comment:
Same here, if it returns then it will not actually invalidate.

Reply:
Yes to avoid the "no image" case after reboot